### DISCUSSION TITLE
Appropriate Boundary Conditions for Pump/Outflow?

### DISCUSSION THREAD

**Discussion by ghost** [shannon-lab/zapdos#121] [Q&A Plasma Transport]
In a couple of my recent discussion posts, I've inquired about using the DoNothingBCs as "open boundaries" only to eventually realize (I think) these boundaries really _do nothing_! 

From what I've gathered recently after browsing through COMSOL's user guide on their plasma module and some other computational works from other groups (Kushner, Raja, etc), that setting up the boundary conditions for either a free boundary or an flow outlet like a pump should be rather simple, but I'm not sure how to implement this in Zapdos. A common one seems to be applying a constant pressure at the outlet boundary and velocity/flux profile at the inlet, but it seems that with the current state of Zapdos and Navier-Stokes not being well-coupled yet makes this approach a bit more challenging at first glance. 

Are there any boundary conditions that would sufficient in describing an outlet such as a pump port in a reactor chamber?

**Comment by csdechant** [comment]
Currently @gsgall is working on coupling Zapdos and Navier-Stokes, such that the heavy species are effected by the background gas flow (this includes introduction dynamic pressure changes during Zapdos runs, while currently Zapdos assumes a constant pressure). 

As of a free boundary (such as a boundary representing open air), we can use the natural boundary conditions (i.e. grad(n) = 0). This BC is the default in MOOSE/Zapdos, so if a user does not define a BC at a boundary, MOOSE/Zapdos sets grad(n) = 0 at that boundary for the variable. 

For your outlet of a pump port in a reactor chamber case, unless the background gas flow is important to the physics you are trying to capture, then the default natural boundary condition should be sufficient. Currently, Zapdos assumes a stagnate background gas at constant pressure and temperature, so the default natural boundary condition is what we currently use for non-wall boundaries.

**Reply by ghost** [reply]
I thought the natural boundaries make flux_n = 0? I apologize if I'm making a mistake.

**Reply by csdechant** [reply]
Yes, that is true for the technical definition for "Natural Boundary Condition" for an undefined operation, so I should clarify. I have often heard the default boundary condition in MOOSE (i.e.  **n**•(∇u)=0) called a natural boundary condition, because it is a natural boundary condition for a purely diffusive problem (i.e. if the strong form is -∇•(D ∇u), then the natural boundary condition is **n**•(∇u)=0). 

So in Zapdos for metastables and potential, which are treated as purely diffusive, MOOSE's default BC is the natural boundary condition. Now for charged species, their strong form is ∇•(μ n ∇V - D ∇n) or ∇•(μ n ∇V) - ∇•(D ∇u), where the first term is advection and the second term is diffusion. For the advection term, our density cannot be zero, so that leaves the BC of **n**•(∇V)=0 (which is the natural boundary condition for the potential) and the diffusion term's natural boundary condition is the default BC in MOOSE. I hope that cleared everything up.
